docs/specification.md

```diff
@@ -646,51 +645,29 @@ Parameter (in Michelson):
 ### **flush**
 
 ```ocaml
-Flush of nat
+type proposal_key = bytes
+
+Flush of [proposal_key]
 ```
 
 Parameter (in Michelson):
 ```
-(nat %flush)
+(bytes list %flush)
 ```
 
-- Finish voting process on an amount of proposals for which the voting period is over.
-- The order of processing proposals are from 'the oldest' to 'the newest'. The proposals which
-have the same timestamp due to being in the same block, are processed in the order of their proposal keys.
+- Finish voting process on proposals that are in the parameters for which the voting period is over.
 - Frozen tokens from voters and proposal submitter associated with those proposals are returned
   in form of tokens in governance token contract:
-  - If proposal got rejected due to the quorum was not met or the quorum was met but upvotes are less then downvotes:
+  - If proposal became expired or got rejected due to the quorum was not met or the quorum was met but upvotes are less then downvotes:
```

Comment:
```suggestion
  - If the proposal expired or got rejected due to the quorum was not met or the quorum was met but upvotes are less then downvotes:
```

I would also add here that a proposal is expired when more than 3 stages have passed since the one where it was created.